File: pset1/mario/mario.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>

void mario_left(int height, int row_num)
{
    for (int i=0; i< (height-row_num); i++)  printf(" ");
    for (int j = 0; j < row_num; j++) printf("#");
}

void mario_right(int height, int row_num)
{
    for (int j = 0; j < row_num; j++) printf("#");
    for (int i=0; i < (height-row_num); i++)printf(" ");
}


int main(void)
{
    
    int height;
    do
    {
      height = get_int("Height: \n");  
    } while (height >8 || height <1);
    
    for (int i=1; i<=height; i++)
        {

            mario_left(height, i);
            printf("  ");

            mario_right(height, i);
            printf("\n");
        }
}

File: pset6/mario/less/mario.py
from cs50 import get_int

def main():
    pyramid_h = get_int("Height: ")
    if (pyramid_h >=1) and (pyramid_h <=8):
        row_height = 1
        for i in range(pyramid_h):
            mario_left(pyramid_h, row_height)
            row_height +=1
        #print("row_height:", row_height)
    else:
        exit()

def mario_left(pyramid_h, row_height):
    for i in range(pyramid_h - row_height):
        #print("checking diff value:", i)
        print(" ", end ="")

    for j in range(row_height):
        #print("Current row_height number:", j)
        print("#", end ="")
    print("\n", end="")

main()